src/Interpreters/Cache/WriteBufferToFileSegment.h

```diff
@@ -16,7 +16,6 @@ class WriteBufferToFileSegment : public WriteBufferFromFileDecorator, public IRe
     explicit WriteBufferToFileSegment(FileSegmentsHolderPtr segment_holder);
 
     void nextImpl() override;
-    ~WriteBufferToFileSegment() override;
```

Comment:
Is it just to be more explicit? As far ad no copy-/move- constuctors defined, it shouldn't be a big difference, we can merge and backport, since PR is ready and add `= default` in another PR, if still needed, wdyt?

Reply:
Since we have virtual functions here it's better to have the destructor defined to avoid possible memory leaks.

Reply:
Since `WriteBufferToFileSegment` inherits from a base class with a virtual destructor, its destructor will still be virtual, ensuring correct cleanup.

https://github.com/ClickHouse/ClickHouse/pull/63403